File: java/dtdgen-master/src/main/java/com/philhanna/dtdgen/dtd/DTDEmptyElement.java
package com.philhanna.dtdgen.dtd;

import com.philhanna.dtdgen.ElementModel;

/**
 * A DTDElementModel that represents an empty element (one that has no
 * children)
 */
public class DTDEmptyElement extends DTDElementModel {

   /**
    * Creates a new empty element over the specified element model
    * @param elementModel the element model
    */
   public DTDEmptyElement(ElementModel elementModel) {
      super(elementModel);
   }

   /**
    * Generates the &lt;!ELEMENT <i>&lt;name&gt;</i> EMPTY&gt;
    */
   @Override
   public String toString() {
      final String output = String
            .format("<!ELEMENT %s EMPTY >", getElementName());
      return output;
   }

}

File: dtdgen/dtd/dtd_empty_element.py
from dtdgen import ElementModel
from dtdgen.dtd import DTDElementModel


class DTDEmptyElement(DTDElementModel):
    """A DTDElementModel that represents an empty element
    (one that has no children)"""

    def __init__(self, element_model: ElementModel):
        """Creates a new empty element for the specified element model"""
        super().__init__(element_model)

    def __str__(self):
        """Generates the <!ELEMENT name EMPTY> string"""
        output = f"<!ELEMENT {self.element_name} EMPTY>"
        return output
